### ISSUE TITLE
Better control over the TableReport's progress display

### ISSUE THREAD

**Issue by jeromedockes** [skrub-data/skrub#1175]
### Problem Description

In some cases the TableReport progress display clutters the output.

- it uses carriage return to update the line it prints. in some cases (eg when stdout is a file, when rendering notebooks with `nbconvert` or generating slides with `quarto`) this just results in many lines which the user may want to avoid. then we may want it to just print an increasing row of "." or similar
- sometimes we may want it to be completely silent
- it should print progress information to stderr, not stdout

### Feature Description

- write progress info to the standard error stream
- maybe add a `verbose` parameter?

**Comment by glemaitre** [comment]
Indeed, I was going to show on such case when creating a jupyter book:

![Uploading image.png…]()

**Comment by reshamas** [comment]
I am working on this: Pearson's correlation coefficient.
skrub/_column_associations.py

via: PyLadiesCon 2024 online sprint

**Comment by rcap107** [comment]
We might want to add rich (https://rich.readthedocs.io/en/stable/) as an optional dependency to deal with this
